server/src/main/java/org/opensearch/search/DocValueFormat.java

```diff
@@ -260,7 +265,14 @@ public String getWriteableName() {
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.writeString(formatter.pattern());
+            if (out.getVersion().before(Version.V_3_0_0) && formatter.equals(DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER)) {
+                out.writeString(DateFieldMapper.LEGACY_DEFAULT_DATE_TIME_FORMATTER.pattern());
+            } else {
+                out.writeString(formatter.pattern());
+            }
+            if (out.getVersion().onOrAfter(Version.V_3_0_0)) {
+                out.writeString(formatter.printPattern());
```

Comment:
```suggestion
                out.writeOptionalString(formatter.printPattern());
```

I believe `printPattern` is not enforced and could be null, right?

Reply:
Thanks for pointing it out